osu.Game/Screens/Play/HUD/SongProgressInfo.cs

```diff
@@ -106,5 +143,12 @@ protected override void Update()
         }
 
         private string formatTime(TimeSpan timeSpan) => $"{(timeSpan < TimeSpan.Zero ? "-" : "")}{Math.Floor(timeSpan.Duration().TotalMinutes)}:{timeSpan.Duration().Seconds:D2}";
+
+        private void keepTextSpritesUpright()
+        {
+            timeCurrentContainer.OnUpdate += Extensions.DrawableExtensions.KeepUprightAndUnscaled;
+            progressContainer.OnUpdate += Extensions.DrawableExtensions.KeepUprightAndUnscaled;
+            timeLeftContainer.OnUpdate += Extensions.DrawableExtensions.KeepUprightAndUnscaled;
```

Comment:
I'm not a huge fan of how this method needs to be attached to drawables via such hooks. I much preferred what #18076 did - namely, it introduced a special container type that would keep itself and its children upright. I would much prefer to use something like that. You could probably even co-opt `GrowToFitContainer` for this (but probably after a rename).

Reply:
Okay i will make the necessary changes, but i explained above GrowToFitContainer has its own purpose, which means that for KeepUprightContainer to be reusable it can't be mixed with GrowToFitContainer.

Reply:
Also what do you think about calling KeepUprightAndUnscaled from inside UprightContainer's update instead of making it a member of UprightContainer so that it can be used in either way?